wsl-builder/prepare-build/build.go

```diff
@@ -224,19 +241,53 @@ func downloadFile(url, dest string) (err error) {
 		return fmt.Errorf("http request failed with code %d", resp.StatusCode)
 	}
 
-	return writeContentInto(resp.Body, dest)
+	var size int
+	if size, err = strconv.Atoi(resp.Header.Get("Content-Length")); err != nil {
+		log.Printf("Warning: unknown size for %s: %v", url, err)
+	}
+	return writeContentInto(resp.Body, uint64(size), dest)
+}
+
+type WriteCounter struct {
+	f               *os.File
+	name            string
+	current         uint64
+	previousPrinted uint64
+	total           uint64
+}
+
+func (wc *WriteCounter) Write(p []byte) (int, error) {
+	n := len(p)
+	wc.current += uint64(n)
+	if wc.current < wc.previousPrinted+10*(1<<20) {
+		return wc.f.Write(p)
+	}
+	log.Printf("%s: %.0f MB / %.0f MB\n", wc.name,
+		math.Floor(float64(wc.current)/(1<<20)),
+		math.Floor(float64(wc.total)/(1<<20)))
+	wc.previousPrinted = wc.current
+	return wc.f.Write(p)
+}
+
+func (wc *WriteCounter) Close() error {
+	return wc.f.Close()
 }
 
 // writeContentInto writes the content of the `source io.Reader`
 // into a new file created at `dest`.
-func writeContentInto(source io.Reader, dest string) (err error) {
+func writeContentInto(source io.Reader, total uint64, dest string) (err error) {
```

Comment:
Doc comments should be updated to indicate the role of the new parameter

Reply:
good one!